src/handlers/process-subscriptions.ts

```diff
@@ -0,0 +1,126 @@
+    airnodeWallet,
+    provider,
+    currentBlock,
+    sponsorAddress
+  );
+
+  // Skip processing for the current sponsorAddress if the wallet functions fail
+  if (isNil(walletData)) {
+    const sponsorLogOptions = buildLogOptions('additional', { sponsor: sponsorAddress }, providerLogOptions);
+    utils.logger.warn('Failed to fetch sponsor wallet or transaction count', sponsorLogOptions);
+    return;
+  }
+
+  const { sponsorWallet, transactionCount } = walletData;
+
+  const sponsorWalletLogOptions = buildLogOptions(
+    'additional',
+    { sponsorWallet: shortenAddress(sponsorWallet.address) },
+    providerLogOptions
+  );
+  utils.logger.logPending(transactionCountLogs, sponsorWalletLogOptions);
+  utils.logger.info(`Processing ${subscriptions.length} subscription(s)`, sponsorWalletLogOptions);
+
+  const processSponsorWalletResult = await processSponsorWallet(
+    airnodeWallet,
+    contracts['DapiServer'],
+    gasTarget,
+    subscriptions,
+    sponsorWallet,
+    voidSigner,
+    transactionCount
+  );
+
+  processSponsorWalletResult.forEach(([logs, data]) => {
+    const subscriptionLogOptions = buildLogOptions('additional', { subscriptionId: data.id }, sponsorWalletLogOptions);
+    utils.logger.logPending(logs, subscriptionLogOptions);
+  });
+};
+
+export const handler = async ({
+  providerSponsorSubscriptions,
+  baseLogOptions,
+}: {
+  providerSponsorSubscriptions: ProviderSponsorSubscriptionsState;
+  baseLogOptions: utils.LogOptions;
+}): Promise<AWSHandlerResponse> => {
+  const airnodeConfig = goSync(loadAirnodeConfig);
+  if (!airnodeConfig.success) {
+    utils.logger.error(airnodeConfig.error.message);
+    throw airnodeConfig.error;
+  }
+
+  const airnodeWallet = ethers.Wallet.fromMnemonic(airnodeConfig.data.nodeSettings.airnodeWalletMnemonic);
+  const provider = node.evm.buildEVMProvider(
+    providerSponsorSubscriptions.providerState.providerUrl,
+    providerSponsorSubscriptions.providerState.chainId
+  );
+  const rrpBeaconServerAbi = new ethers.utils.Interface(protocol.RrpBeaconServerFactory.abi).format(
+    ethers.utils.FormatTypes.minimal
+  );
+
+  const dapiServerAbi = [
+    'function conditionPspBeaconUpdate(bytes32,bytes,bytes) view returns (bool)',
+    'function fulfillPspBeaconUpdate(bytes32,address,address,address,uint256,bytes,bytes)',
+  ];
+
+  const abis: { [contractName: string]: string | string[] } = {
+    RrpBeaconServer: rrpBeaconServerAbi,
+    DapiServer: dapiServerAbi,
+  };
+  const contracts = Object.entries(providerSponsorSubscriptions.providerState.chainConfig.contracts).reduce(
+    (acc, [contractName, contractAddress]) => {
+      if (isNil(abis[contractName])) {
+        return acc;
+      }
+      return { ...acc, [contractName]: new ethers.Contract(contractAddress, abis[contractName], provider) };
+    },
+    {}
+  );
+  const voidSigner = new ethers.VoidSigner(ethers.constants.AddressZero, provider);
```

Comment:
I moved this out and called the function `reinitializeProvider`, but I guess it would be more accurate to refactor the current `initializeProvider` function to be called `initializeEvmState` (or something similar) and call this function `initializeProvider`. Do you think this is necessary?

Reply:
Sure, I agree with that :+1: